assets/js/googlesitekit/modules/datastore/modules.js

```diff
@@ -230,6 +233,7 @@ const baseActions = {
 		order,
 		homepage,
 		settingsComponent = DefaultModuleSettings,
+		setupComponent = DefaultModuleSetup,
```

Comment:
Revisiting this, we may want to diverge from the ACs here in that there should not be any default value for the `setupComponent` argument. Per #1623, this will be changed for the `settingsComponent` replacements as well. Providing a default here prevents us from checking whether the module has a setup component attached or not - which will come in handy in the future, e.g. when we need to determine whether we need to link to the module setup screen or the regular module page (linking to the setup screen when there is no setup component would not make sense). There is currently some hard-coded exception in place for PageSpeed Insights (which doesn't require a setup) for that very reason - which we can eliminate by checking whether the module has a setup component.

Reply:
Yeah, I think that makes sense. A `DefaultModuleSetup` component should be capable of setting up the module, but setup is very module-specific so I don't think it's possible to have such a thing in our case as we can't really assume anything about how every module works to have a generic setup component.

`setupComponent` can default to `null` if not passed and is what I have in #1623 for settings components so far (almost ready for review, but you can see it here: https://github.com/google/site-kit-wp/pull/2202/files#diff-51079bbb133e20fa794c0e1fd5fd30b551dda818e0cfe0ba4162502670cf133a)